server/utils/validation/auth.js

```diff
@@ -0,0 +1,26 @@
+const { object, string, ref } = require('yup');
+
+const registerValidation = async (req, res, next) => {
+  const schema = object().shape({
+    email: string().min(10).required(),
+    password: string().min(8).required(),
+    confirmPassword: string()
+      .oneOf([ref('password'), null], 'Passwords must match')
+      .required(),
+    firstName: string().min(3).required(),
+    lastName: string().min(3).required(),
+    role: string().required(),
+  });
+  try {
+    await schema.validate(req.body, { abortEarly: false });
```

Comment:
Just make schema object here then validate schema at controller itself, That's how they worked (Rana's team)